src/components/AddApplication/AddApplicationSchema.js

```diff
@@ -31,32 +33,140 @@ export const ApplicationSummary = () => (<TextContent>
     </Text>
 </TextContent>);
 
+export const SelectAuthenticationDescription = ({ applicationTypeName, authenticationTypeName }) => (
+    <TextContent>
+        <Text component={ TextVariants.p }>
+            <FormattedMessage
+                id="sources.selectAuthenticationDescription"
+                defaultMessage="Selected application { applicationTypeName } supports { authenticationTypeName } authentication
+                type. You can use already defined authentication values or define new."
+                values={{ applicationTypeName, authenticationTypeName }}
+            />
+        </Text>
+    </TextContent>
+);
+
+SelectAuthenticationDescription.propTypes = {
+    applicationTypeName: PropTypes.string,
+    authenticationTypeName: PropTypes.string
+};
+
 export const hasAlreadySupportedAuthType = (authValues = [], appType, sourceTypeName) =>
     authValues.find(({ authtype }) => authtype === get(appType, `supported_authentication_types.${sourceTypeName}[0]`));
 
+export const selectAuthenticationStep = ({
+    intl, source, authenticationValues, sourceType, applicationTypes, modifiedValues
+}) => {
+    const nextStep = ({ values: { application } }) => {
+        const app = application ? application : {};
+        const appId = app.application_type_id ? app.application_type_id : '';
+
+        return `${sourceType.name}-${appId}`;
+    };
+
+    const fields = [{
+        component: 'description',
+        name: 'authtypesetter',
+        Content: AuthTypeSetter,
+        authenticationValues,
+        hideField: true,
+        modifiedValues
+    }];
+
+    applicationTypes.forEach((app) => {
+        const supportedAuthType = get(app, `supported_authentication_types[${sourceType.name}][0]`, '');
+
+        if (supportedAuthType && hasAlreadySupportedAuthType(authenticationValues, app, sourceType.name)) {
+            const supportedAuthTypeName =
+                get(sourceType, `schema.authentication`, {}).find(({ type }) => type === supportedAuthType).name;
+
+            const radioOptions = authenticationValues.filter(({ authtype }) => authtype === supportedAuthType).map((values) => {
+                const app = source.applications.find(({ authentications }) => authentications.find(({ id }) => id === values.id));
+
+                const appTypeId = app ? app.application_type_id : '';
```

Comment:
No need to assign this to a value
```JS
const appType = applicationTypes.find(({ id }) => id === app && app.application_type_id) ||
  {
    display_name: `unused-${values.id}`
  }
const includeUsername = values.username ? `-${values.username}` : '';

  return {
    label: `${supportedAuthTypeName}${includeUsername}-${appType.display_name}`
  }
```

You could simplify it even more by directly calculating `includeUsername` and `appType` when creating label but that is kinda hard to read.
```JS
return {
    label: `${supportedAuthTypeName}${
      values.username ? `-${values.username}` : ''
    }-${(
      applicationTypes.find(({ id }) => id === app && app.application_type_id) ||
     {
       display_name: `unused-${values.id}`
     }
    ).display_name}`
  }
```